Platform/ApollolakeBoardPkg/Library/Stage1BBoardInitLib/Stage1BBoardInitLib.c

```diff
@@ -459,11 +460,37 @@ ModuleIdInitialize (
     }
 
     break;
+
+  case PLATFORM_ID_UP2:
+    ModuleId = 0;
+    for (Index = 0; Index < ARRAY_SIZE (mUp2ModuleIdOffset); Index ++) {
+      Offset = GPIO_PADBAR + mUp2ModuleIdOffset[Index];
+      PadConfg0.padCnf0 = GpioRead (NORTHWEST, Offset + BXT_GPIO_PAD_CONF0_OFFSET);
```

Comment:
If use macro NW_GPIO_214/NW_GPIO_215, the code will be:
 Offset = mUp2ModuleIdOffset[Index] & 0xFFFF;
 PadConfg0.padCnf0 = GpioRead ((mUp2ModuleIdOffset[Index] >> 16), Offset + BXT_GPIO_PAD_CONF0_OFFSET);
Similar for other GpioRead/GpioWrite.

Reply:
I followed GPMRB  SKU detection code. The APL GPIO code requires refactor in one shot.